users/serializers.py

```diff
@@ -41,9 +41,13 @@
 USERNAME_ALREADY_EXISTS_MSG = (
     "A user already exists with this username. Please try a different one."
 )
+EMAIL_CONFLICT_MSG = (
+    "This email is associated with an existing account. Please try a different one."
+)
 
-OPENEDX_USERNAME_VALIDATION_MSGS_MAP = {
-    "It looks like this username is already taken": USERNAME_ALREADY_EXISTS_MSG
+OPENEDX_ACCOUNT_CREATION_VALIDATION_MSGS_MAP = {
+    "It looks like this username is already taken": USERNAME_ALREADY_EXISTS_MSG,
+    "This email is already associated with an existing account": EMAIL_CONFLICT_MSG,
```

Comment:
I think we should change this to make it simple:
```suggestion
    "username": USERNAME_ALREADY_EXISTS_MSG,
    "email": EMAIL_CONFLICT_MSG,
```

Reply:
And then this would have impact on the usage.